src/SummaryRow.tsx

```diff
@@ -20,12 +20,11 @@ interface SummaryRowProps<R, SR> extends SharedRowRendererProps<R, SR> {
 }
 
 const summaryRow = css`
-  &.${row} {
+  z-index: 3;
+  line-height: var(--rdg-summary-row-height);
+
+  &.${row} > .${cell} {
```

Comment:
Isn't this good enough?
```suggestion
  & > .${cell} {
```

Reply:
https://github.com/adazzle/react-data-grid/pull/2630